infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -431,7 +434,11 @@ private void waitMachines(
                 }
                 serversAndProbesFuture.complete(null);
 
-                tracerUtil.finishSpan(tracingSpan);
+                if (ex == null) {
```

Comment:
If ex is null function returns at line 426, so this should be an unreachable code